python/paddle/fluid/tests/unittests/dygraph_to_static/test_word2vec.py

```diff
@@ -0,0 +1,325 @@
+                        label_batch).astype("float32"), np.array(
+                            eval_word_batch).astype("int64")
+                center_word_batch = []
+                target_word_batch = []
+                label_batch = []
+                eval_word_batch = []
+
+    if len(center_word_batch) > 0:
+        yield np.array(center_word_batch).astype("int64"), np.array(
+            target_word_batch).astype("int64"), np.array(label_batch).astype(
+                "float32"), np.array(eval_word_batch).astype("int64")
+
+
+#for _, batch in zip(range(10), build_batch(dataset, 128, 3)):
+#    print(batch)
+
+
+class SkipGram(fluid.dygraph.Layer):
+    def __init__(self, name_scope, vocab_size, embedding_size, init_scale=0.1):
+        super(SkipGram, self).__init__(name_scope)
+        self.vocab_size = vocab_size
+        self.embedding_size = embedding_size
+
+        self.embedding = Embedding(
+            size=[self.vocab_size, self.embedding_size],
+            dtype='float32',
+            param_attr=fluid.ParamAttr(
+                name='embedding_para',
+                initializer=fluid.initializer.UniformInitializer(
+                    low=-0.5 / self.embedding_size,
+                    high=0.5 / self.embedding_size)))
+
+        self.embedding_out = Embedding(
+            size=[self.vocab_size, self.embedding_size],
+            dtype='float32',
+            param_attr=fluid.ParamAttr(
+                name='embedding_out_para',
+                initializer=fluid.initializer.UniformInitializer(
+                    low=-0.5 / self.embedding_size,
+                    high=0.5 / self.embedding_size)))
+
+    @declarative
+    def forward(self, center_words, target_words, label):
+        center_words_emb = self.embedding(center_words)
+        target_words_emb = self.embedding_out(target_words)
+
+        # center_words_emb = [batch_size, embedding_size]
+        # target_words_emb = [batch_size, embedding_size]
+        word_sim = fluid.layers.elementwise_mul(center_words_emb,
+                                                target_words_emb)
+        word_sim = fluid.layers.reduce_sum(word_sim, dim=-1)
+
+        pred = fluid.layers.sigmoid(word_sim)
+
+        loss = fluid.layers.sigmoid_cross_entropy_with_logits(word_sim, label)
+        loss = fluid.layers.reduce_mean(loss)
+
+        return pred, loss
+
+
+batch_size = 512
+epoch_num = 1
+embedding_size = 200
+learning_rate = 1e-3
+total_steps = len(dataset) * epoch_num // batch_size
+
+
+def get_similar_tokens(query_token, k, embed):
+    W = embed.numpy()
+    x = W[word2id_dict[query_token]]
+    cos = np.dot(W, x) / np.sqrt(np.sum(W * W, axis=1) * np.sum(x * x) + 1e-9)
+    flat = cos.flatten()
+    indices = np.argpartition(flat, -k)[-k:]
+    indices = indices[np.argsort(-flat[indices])]
+    for i in indices:  # Remove the input words
+        print('for word %s, the similar word is %s' %
+              (query_token, str(id2word_dict[i])))
```

Comment:
It seems that `get_similar_tokens ` is not used in the unittest.

Reply:
This is copied code so I didn't care much about this. The get_similar_tokens is used in the original code but it now cannot run because of API compatibility issue. I will remove it in this unittest.